src/ppx/xr_composition_layers.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if defined(PPX_BUILD_XR)
+#include "ppx/xr_composition_layers.h"
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+namespace ppx {
+
+XrProjectionLayer::XrProjectionLayer()
+{
+    layer().views = mViews.data();
+}
+
+void XrProjectionLayer::AddView(XrCompositionLayerProjectionView view)
+{
+    mViews.emplace_back(view);
+    layer().views     = mViews.data();
+    layer().viewCount = static_cast<uint32_t>(mViews.size());
+}
+
+void XrProjectionLayer::AddView(XrCompositionLayerProjectionView view, XrCompositionLayerDepthInfoKHR depthInfo)
+{
+    view.next = &depthInfo;
```

Comment:
This doesn't seem right, did you mean to get a reference to the object emplaced back on line 40 instead?